share/lib/python/neuron/hclass3.py

```diff
@@ -4,38 +4,116 @@
 # h.anyclass methods may be overridden. If so the base method can be called
 # using the idiom self.basemethod = self.baseattr('methodname')
 # ------------------------------------------------------------------------------
-      
-from neuron import h, hoc
+
+__all__ = ["hclass", "nonlocal_hclass", "HocBaseObject"]
+
+from . import h, hoc
 import nrn
+import sys
+
+def assert_not_hoc_composite(cls):
+    """
+    Asserts that a class is not directly composed of multiple HOC types.
+    """
+    hoc_bases = [b for b in cls.__bases__ if issubclass(b, hoc.HocObject)]
+    if len(hoc_bases) > 1:
+        bases = ", ".join(b.__name__)
+        cname = cls.__name__
+        raise TypeError(f"Composition of {bases} HocObjects not allowed in {cname}")
```

Comment:
Note: f-strings require Python 3.6+

Reply:
I see... Python 3.5 goes through a different file.

Reply:
Yea, Python 3.5 or earlier just keeps using the old `hclass3.py` file since the required improvements to metaclasses were simply not available in earlier versions, so I kept things the same there.